Test failures caused by a changed wording of failure messages

## Expected Behavior
The build and tests succeed when checking out `origin/main`

## Observed Behavior
The build fails with the following message:

```
    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':detektMain'.
    > Analysis failed with 3 weighted issues.

    * Try:
    > Run with --stacktrace option to get the stack trace.
    > Run with --info or --debug option to get more log output.
    > Run with --scan to get full insights.
    > Get more help at https://help.gradle.org.

    BUILD FAILED in 29s
    3 actionable tasks: 3 executed
    "
    to contain:
      "failed with 3 issues." 
        at io.gitlab.arturbosch.detekt.JvmSpec.Type resolution on JVM(JvmSpec.kt:17)
```

The full build log is attached here: [buildLog.txt](https://github.com/detekt/detekt/files/13635781/buildLog.txt)

Amongst others, it fails at `JvmSpec.kt` line 17 which states:

```kotlin
assertThat(result.output).contains("failed with 3 issues.")
```

The actual output, however, contains `"failed with 3 weighted issues."`, so, changing the above line to 

```kotlin
assertThat(result.output).contains("failed with 3 weighted issues.")
```

fixes the issue. I will shortly post a PR that contains all places where this is happening.

## Steps to Reproduce
1. Clone the repo
2. Checkout `origin/main`  (In particular, I ran this on `8e350ee7a895baacdbeb4480812cfb173ae750a8`)
3. Run `./gradlew build -x dokkaHtml`

## Context
N/A

## Your Environment
* Version of detekt used: N/A
* Commit affected: `8e350ee7a895baacdbeb4480812cfb173ae750a8`

* Version of Gradle used (if applicable): 8.5
* Gradle scan link (add `--scan` option when running the gradle task): https://gradle.com/s/bwo67jnwume24
* Operating System and version: Windows 10 Version 21H2
* Link to your project (if it's a public repository): N/A

--- a/.github/CONTRIBUTING.md
+++ b/.github/CONTRIBUTING.md
@@ -1,6 +1,7 @@
 # Contributing to detekt
 
 - Read [this article](https://chris.beams.io/posts/git-commit/) before writing commit messages.
+- Before running any tests, run `./gradlew publishToMavenLocal` to ensure that all tests are using the locally built artifact (c.f. issue [#6708](https://github.com/detekt/detekt/issues/6708) and PR [#6415](https://github.com/detekt/detekt/pull/6415)). Otherwise, the tests will pull a possibly outdated version of detekt from Maven Central or Sonatype.
 - Use `gradle build -x dokkaHtml` to build the source but exclude documentation JAR generation to save time.
 - Make sure that `gradle detekt` does not report any errors.
 - This repository follows the [Kotlin coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html),

